src/Components/Guestbook/Guestbook.jsx

```diff
@@ -0,0 +1,66 @@
+import { useState } from 'react';
+import { useInputs } from '../../Context-Hooks/InputContext';
+import { useUser } from '../../Context-Hooks/UserContext';
+
+export default function Guestbook() {
+    // create state + name + input text / allow contexts
+    const [name, setName] = useState('');
+    const [userMessage, setUserMessage] = useState('');
+    const {inputs, setInputs} = useInputs();
+    const {user, setUser} = useUser();
+
+    function updateUserName() {
+        if(!userMessage) return;
+
+        setUser(name);
+        setInputs([...inputs, {name, message: userMessage}]);
+        setUserMessage('');
+    }
+
+    const submitHandler = (e) => {
+        e.preventDefault();
+        updateUserName()
+    }
+
+    const userNameInput = (
+        <section>
+            <label htmlFor='name'>Guest Name: </label>
+            <input type='text' placeholder='Enter Name..' id='name' value={name} onChange={(e) => setName(e.target.value)}></input>
```

Comment:
Remember that empty inputs should always be self-closing. This won't break anything, but semantically and stylistically it is more common.